### ISSUE TITLE
translating       TaskSpec to Pod: error getting image manifest

### ISSUE THREAD

**Issue by ycyxuehan** [tektoncd/pipeline#3655]
version: v0.17.2
taskrun:
```yaml
  podTemplate:
    dnsConfig:
      nameservers:
      - 223.5.5.5
      - 223.6.6.6
      - 114.114.114.114
    imagePullSecrets:
    - name: demo-registry
  resources:
    inputs:
    - name: demo
      resourceRef:
        name: demo
    outputs:
    - name: resourceImage
      resourceRef:
        name: demo-resource-image
  serviceAccountName: demo
....
```
service account:
```yaml
secrets:
- name: demo
- name: demo-git-token
- name: demo-git-privatekey
- name: demo-registry
- name: demo-token-gzcjp
```
secrets:

```yaml
apiVersion: v1
data:
  .dockerconfigjson: eyJhdXRocyI6eyJyZWd...
kind: Secret
metadata:
  annotations:
    tekton.dev/docker-0: https://demo.myregistry.com
type: kubernetes.io/dockerconfigjson
```

**Comment by sbwsg** [comment]
@ycyxuehan Does this remain an issue for you?

**Comment by shokohsc** [comment]
Hello, i'm still having this issue as of
gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/controller:v0.29.0@sha256:72f79471f06d096cc53e51385017c9f0f7edbc87379bf415f99d4bd11cf7bc2b
installed from tektoncd/operator and only when the task's step args key is defined, if I add the command key too, it goes flawlessly.

Relates to:
- https://github.com/tektoncd/pipeline/issues/2707 
- https://github.com/tektoncd/pipeline/issues/3105
There's a workaround when you're running without the operator installing the pipelinescontroller: 
https://github.com/tektoncd/pipeline/issues/1539#issuecomment-640457021
How would one do it with the operator installing the pipelinescontroller ?!

**Comment by imjasonh** [comment]
If the image is private and requires credentials, those credentials must be available to the `tekton-pipelines-controller` SA (the SA that runs the Tekton controller) as an `imagePullSecret`. The Tekton controller uses this to get image metadata to determine what command to run when a step starts, which it only needs to do when the step spec doesn't explicitly specify it.

**Comment by shokohsc** [comment]
I'm having the same error after adding imagePullSecret to `tekton-pipelines-controller` service account. Guess I'll need to define every command from now on.

**Comment by shokohsc** [comment]
Reverting to 0.23.0 works too

**Comment by shokohsc** [comment]
I think this will prevent from using our credentials via an imagePullSecret: https://github.com/tektoncd/pipeline/commit/49a7fa2db63fc633190da260d7167c05a8a7a91d
It's still there as of today: https://github.com/tektoncd/pipeline/blob/main/config/controller.yaml#L99
But commit message says it's needed Oo
I'm lost. Would there be a way to delete those environment variables from the operator ? Asking for a friend :)

**Comment by markbastiaans** [comment]
Same issue here, related to #2707.

We're running v0.30.0 on EKS and are running into this issue with ECR. Prior to this we were on v0.19.x.
I've currently managed to work around the issue by copy-pasting entrypoint commands. Deleting the dummy env vars from our (custom) Helm chart seemed kind of a big risk to me with the Go AWS SDK not being updated yet / no structural fix known yet for #4087.

A structural solution would be nice.

**Comment by sbwsg** [comment]
If you're running in AWS I would expect (tho can't confirm) that removing the env vars should be fine. We're slowly making progress on #4087 but I'm having a bit of trouble compiling and testing with the update that brings in the newer go aws sdk.
